src/checkpatch_wrapper.sh

```diff
@@ -10,17 +12,41 @@ function execute_checkpatch()
   # --codespell $external_script_path/spelling.txt
 
   local -r options="--terse --no-tree --color=always -strict --file "
-  local -r checkpatch="perl $external_script_path/checkpatch.pl $options"
+  local -r script="scripts/checkpatch.pl $options"
+  local kernel_root=""
+  local -r original_working_dir=$PWD
 
   # Check if is a valid path
-  if [ ! -d $FILE_OR_DIR_CHECK -a ! -f $FILE_OR_DIR_CHECK ]; then
+  # if [ ! -d $FILE_OR_DIR_CHECK -a ! -f $FILE_OR_DIR_CHECK ]; then
+  #   complain "Invalid path"
+  #   return 1
+  # fi
+  if [[ -z $FILE_OR_DIR_CHECK ]]; then
+    FILE_OR_DIR_CHECK="."
+  fi
+
+  FILE_OR_DIR_CHECK="$(realpath $FILE_OR_DIR_CHECK)"
+  # Check if is a valid path
+  if [[ ! -d $FILE_OR_DIR_CHECK && ! -f $FILE_OR_DIR_CHECK ]]; then
     complain "Invalid path"
     return 1
   fi
 
+  # try to find kernel root at given path
+  kernel_root="$(find_kernel_root $FILE_OR_DIR_CHECK)"
+  if [[ -z "$kernel_root" ]]; then
+    # fallback: try to find kernel root at working path
+    kernel_root="$(find_kernel_root $original_working_dir)"
+  fi
+
+  # Check if kernel root was found.
+  if [[ -z "$kernel_root" ]]; then
+    complain "Neither the given path nor the working path is in a kernel tree."
+    return 1
+  fi
+  
```

Comment:
Nitpick: in the `git log` I can seen an extra space in the line 47, just remove it.